sys/net/gnrc/network_layer/ipv6/nib/_nib-router.c

```diff
@@ -191,11 +191,11 @@ static gnrc_pktsnip_t *_build_ext_opts(gnrc_netif_t *netif,
             }
         }
     }
-    ltime = (gnrc_netif_is_6lbr(netif)) ?
-            (SIXLOWPAN_ND_OPT_ABR_LTIME_DEFAULT) :
-            (abr->valid_until - evtimer_now_min());
-    (void)ltime;    /* gnrc_sixlowpan_nd_opt_abr_build might evaluate to NOP */
-    abro = gnrc_sixlowpan_nd_opt_abr_build(abr->version, ltime, &abr->addr,
+    ltime_min = (gnrc_netif_is_6lbr(netif)) ?
+                (SIXLOWPAN_ND_OPT_ABR_LTIME_DEFAULT) :
+                ((abr->valid_until_ms - evtimer_now_msec()) / MS_PER_SEC / SEC_PER_MIN);
```

Comment:
might as well
```suggestion
                gnrc_ipv6_nib_abr_valid_offset(abr);
```

Reply:
```
expected ‘const gnrc_ipv6_nib_abr_t *’ but argument is of type ‘_nib_abr_entry_t *’
  117 | static inline uint32_t gnrc_ipv6_nib_abr_valid_offset(const gnrc_ipv6_nib_abr_t *abr)
```

Reply:
arg those are just copy and paste :weary: 

I guess you could add a `_nib_abr_entry_valid_offset()` then - another PR could then change `_nib_abr_entry_t` to

```C
typedef struct {
    gnrc_ipv6_nib_abr_t super;
    evtimer_msg_event_t timeout;    /**< timeout of the information */
    /**
     * @brief   Bitfield marking the prefixes in the NIB's off-link entries
     *          disseminated by _nib_abr_entry_t::addr
     */
    BITFIELD(pfxs, CONFIG_GNRC_IPV6_NIB_OFFL_NUMOF);
    /**
     * @brief   Bitfield marking the contexts disseminated by
     *          _nib_abr_entry_t::addr
     */
    BITFIELD(ctxs, GNRC_SIXLOWPAN_CTX_SIZE);
} _nib_abr_entry_t;
```

Reply:
missing some inheritance / function templates / duck-typing in C-lang

Reply:
i am not sure if a one use function is needed here?

Reply:
It would surely be cleaner and make the code easier to read